cognee/api/v1/search/search.py

```
@@ -175,6 +176,14 @@ async def search(
         if not datasets:
             raise DatasetNotFoundError(message="No datasets found.")
 
+    graph_engine = await get_graph_engine()
+    nodes_count = await graph_engine.count_nodes()
+
+    if nodes_count == 0:
+        raise SearchOnEmptyGraphError(
```

Do we want to raise the exception? I feel its good but just to point it out. If the graph is empty the search will fail